### PR TITLE
Display argument values if they mismatch

### PR DESCRIPTION
**Pull request by Eluss**
Hi there,
this PR provides more detailed messages in case of arguments mismatch. It should cover tasks #13 and #30. WDYT?

### CODE REVIEW

File: Mimus/Source/Verification/MismatchMessageCreator.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import Foundation
+
+class MismatchMessageCreator {
+    
+    func message(for mismatchedResults: [MatchResult]) -> String {
+        guard !mismatchedResults.isEmpty else {
+            return ""
+        }
+        let mismatchedCallsCountInfo = "Found \(mismatchedResults.count) call(s) with expected identifier, but not matching arguments:"
```
Comment (paweldudek) [inline Mimus/Source/Verification/MismatchMessageCreator.swift]: This gets a bit too wordy IMO in final message:

```
failed - Call with identifier Identifier was recorded 1 times, but arguments didn't match. Found 1 call(s) with expected identifier, but not matching arguments:
Mismatched call #1:
Mismatch in argument #2 - expected <(["type": "logina"])>, but was <(["type": "login"])>.
```

We're now basically repeating the same message twice. Verifying for a single a call will be the most commonly used scenario so I'd make the message as short as possible. Ideally this should state:

```
failed - Call with identifier Identifier was recorded 1 times, but arguments are different.

Mismatch in argument #2 - expected <(["type": "logina"])>, but was <(["type": "login"])>.
```

And for longer messages we currently have:

```
failed - Call with identifier was recorded 1 times, but expected 2. Found 1 call(s) with expected identifier, but not matching arguments:
Mismatched call #1:
Mismatch in argument #0 - expected <(Fixture Email)>, but was <(Fixture Emaila)>.
```

I'd alter this to (more concise):

```
failed - Call with identifier was recorded 1 times, but expected 2. 1 additional call(s) matched identifier, but not arguments:
Mismatched call #1:
Mismatch in argument #0 - expected <(Fixture Email)>, but was <(Fixture Emaila)>.
```

File: Mimus/Source/Verification/MismatchMessageCreator.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    
+    private func displayText(for value: Any?) -> String {
+        if let value = value {
+            return "<(\(String(describing: value)))>"
+        } else {
+            return "<(nil)>"
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (Eluss) [reply Mimus/Source/Verification/MismatchMessageCreator.swift]: Do you mean the lack of it? That's pretty interesting as I indeed cannot see the newline on github page (in any files) but it is clearly there when I check it on any desktop editor. Do you know the possible reason for this?

File: Mimus/Source/Verification/MismatchMessageCreator.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
+//
+//  Copyright © 2019 AirHelp. All rights reserved.
+//
+
+import Foundation
+
+class MismatchMessageCreator {
```
[the remark this answers is not in the document]
Comment (Eluss) [reply Mimus/Source/Verification/MismatchMessageCreator.swift]: Changed :+1:

File: Mimus/Source/Verification/MismatchMessageCreator.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    
+    func message(for mismatchedResults: [MatchResult]) -> String {
+        guard !mismatchedResults.isEmpty else {
+            return ""
+        }
+        let mismatchedCallsCountInfo = "Found \(mismatchedResults.count) call(s) with expected identifier, but not matching arguments:"
+        return mismatchedResults
+            .enumerated()
+            .map {[unowned self] (index, result) -> String in
```
[the remark this answers is not in the document]
Comment (Eluss) [reply Mimus/Source/Verification/MismatchMessageCreator.swift]: I've just checked it and you're right. Thanks for pointing this out 👍

File: Mimus/Source/Verification/MismatchMessageCreator.swift (hunk 1 above)
Comment (Eluss) [reply Mimus/Source/Verification/MismatchMessageCreator.swift]: I've made the changes to the messages that you mentioned. I'm not sure if I got it exactly the way that you noted, but I think that it should cover mentioned issues.
1. If it turns out that there is only one recorded call, then we do not add the `Mismatched call #1:` messages above the listed arguments as there is no point in this.  

2. In case of longer messages, we will have the `1 additional call(s) matched identifier, but not arguments:` message.

3. I still have a small issue with assigning numbers to mismatched arguments as I feel that counting from 0 in this case might not be that intuitive. WDYT?

Comment (paweldudek) [reply Mimus/Source/Verification/MismatchMessageCreator.swift]: It's perfect now. Merging! 🚀 

One minor nitpick I'll add manually is a bit of newline spacing to get the message a bit easier to parse using eyes. 

As per point 3. I think it's better to have 0-indexed indexing as it's a thing that programmers are more accustomed to.
